### ISSUE TITLE
Better organize "Currently installed Apps"-page

### ISSUE THREAD

**Issue by Anduh** [SideQuestVR/SideQuest#101]
### Current Situation
The list is showing all installed apps in alphabetical order, where they are listed by their package-name, unless they have an entry in [SideQuestRepos/app-index.json](https://github.com/the-expanse/SideQuestRepos/blob/master/app-index.json)

There are just over 100 entries in that list atm, while here are 1000s of apks available just through SideQuest alone. So basically 95% of installed apps are listed under C, due to the default `com.`-prefix on package names, which leads to most apps listed on [app-index.json](https://github.com/the-expanse/SideQuestRepos/blob/master/app-index.json) to appear _after_ all the random apps that don't have proper names/icons, which is kinda dumb. 

**Example** How the page looks for me, vs how it could look like:
![side-long](https://user-images.githubusercontent.com/2890634/104848997-2751ce00-58f0-11eb-9005-be735c1be434.png)

### Proposal
It would be sensible to have all the Apps with Images/Proper Names listed alphabetically, first, and _then_ list the all the other apps installed that just have their package-name. This would make the **Currently installed Apps**-list more useful by listing first the tiny list of more prominent apps first(apps anyone have bothered making an entry for in [app-index.json](https://github.com/the-expanse/SideQuestRepos/blob/master/app-index.json) ), and then the rest. This fix could later be extended/upgraded later to list SideQuest-published apps before apps from random sources.

### How it currently works
* [packages.component.ts](https://github.com/SideQuestVR/SideQuest/blob/master/desktop-app/src/app/packages/packages.component.ts) contains [isConnected](https://github.com/SideQuestVR/SideQuest/blob/f34200776c1ae55b36b771a7fa6b9927d92a1ee2/desktop-app/src/app/packages/packages.component.ts#L116), which seems to manages how things are listed
* it calls [getPackages](https://github.com/SideQuestVR/SideQuest/blob/f34200776c1ae55b36b771a7fa6b9927d92a1ee2/desktop-app/src/app/adb-client.service.ts#L172), that uses `adbCommand` to check all packages installed, and alternatively filters oculus/systems apps, sorts list
* then it calls [RepoService](https://github.com/SideQuestVR/SideQuest/blob/master/desktop-app/src/app/repo.service.ts) to pull info from [SideQuestRepos/app-index.json](https://github.com/the-expanse/SideQuestRepos/blob/master/app-index.json) to see if it should override the name/image of any package installed
* then it sorts the list again alphabeticly
* the app icons are pulled from the [app-index-icons](https://github.com/the-expanse/SideQuestRepos/tree/master/app-index-icons) folder

### (Possible) ways to fix/update it
(not familiar with how SideQuest/TypeScript works, but here are a few way one could attempt to improve the situation)
**1. app-index listed first:** update `isConnected` to separate apps found on `app-index.json` and list them alphabetically, then list all other apps alphabetically.
**2. Simple Name parsing 1:**  Instead of [simply returning the packagename](https://github.com/SideQuestVR/SideQuest/blob/f34200776c1ae55b36b771a7fa6b9927d92a1ee2/desktop-app/src/app/packages/packages.component.ts#L133) when it isn't listed in `app-index`, parse the names of all packages, (or the list after index is applied) and remove `com.` from the start. Now apps are listed at least by their creator's name, rather than all starting with `com.`
**3. Simple Name parsing 2:** parse all package names, and display by default the app under the name after the last `.`, if the app isn't on app-index. (packages usually have the form `com.NameOfCreator.NameOfApp`, so anything after the last `.` is usually a variation of the app's name)
**4. All SideQuest Apps:** pull info from somewhere on all the apps listed on SideQuest. This App database/list should contain the app's name, and their package name, so this info could be used to either output a list similar to the `app-index`-format, which then could lead to show any package listed on the SideQuest to show by their proper name(and maybe show the SideQuest-logo as deafult).  This would then be used to list all sidequest apps in alphabetical order with the apps name and either their own logo or the SideQuest placeholder, and then list all unlisted packages at the bottom.

I might attempt at implementing 1, 2, or 3 after having learned more about things, unless someone creates a fix for it right away.

**Comment by braedongough** [comment]
Hey @Anduh - just put up a potential fix for this. It's a rather simple approach, so I'm open to expanding on it. 

I think in a future PR, I would be happy to implement a small search bar to make it easier to search through all installed games.

**Comment by Anduh** [comment]
Nice, a first step in the right direction!

The other options I suggested, (such as parsing any starting `com.` from the packagenames) could be applied to what is shown as the name, which would make the list more readable. Alternatively, with what you've already done would be simply changed to set the displayed names as `NameOfApp (com.NameOfCreator.NameOfApp)`, to keep it both redadable, as well as preserve the package name, in case what's after the last `.` isn't the proper App name, like is the case with a good bunch of demos and games people put together by themselves. Not even SideQuest requires the packages to actually match the app's name. ([Battle Talent](https://sidequestvr.com/app/2348/battle-talent) is one example of one game that had a competently unrelated package name on SideQuest, which was hard to spot until I made an entry for it on the `app-index`.

> I think in a future PR, I would be happy to implement a small search bar to make it easier to search through all installed games.

There already exists a search-bar, are you sure you have the latest version of Sidequest?

![SQsearch](https://user-images.githubusercontent.com/2890634/105607410-295fd500-5da7-11eb-9c09-76b025996aa6.png)
